pkg/testutils/eventually_test.go

```diff
@@ -0,0 +1,78 @@
+package testutils
+
+import (
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEventually_ImmediateSuccess(t *testing.T) {
+	start := time.Now()
+	result := Eventually(t, func() bool {
+		return true
+	}, 1*time.Second, 10*time.Millisecond)
+	elapsed := time.Since(start)
+
+	assert.True(t, result, "Should return true when condition is immediately true")
+	assert.Less(t, elapsed, 50*time.Millisecond, "Should return immediately without waiting")
+}
+
+func TestEventually_EventualSuccess(t *testing.T) {
+	var counter atomic.Int32
+
+	result := Eventually(t, func() bool {
+		count := counter.Add(1)
+		return count >= 3
+	}, 1*time.Second, 10*time.Millisecond)
+
+	assert.True(t, result, "Should return true when condition eventually becomes true")
+	assert.GreaterOrEqual(t, counter.Load(), int32(3), "Should have checked condition at least 3 times")
+}
+
+func TestEventually_Timeout(t *testing.T) {
+	var counter atomic.Int32
+	start := time.Now()
+
+	result := Eventually(t, func() bool {
+		counter.Add(1)
+		return false
+	}, 50*time.Millisecond, 10*time.Millisecond)
+	elapsed := time.Since(start)
+
+	assert.False(t, result, "Should return false when condition never becomes true")
+	assert.GreaterOrEqual(t, elapsed, 50*time.Millisecond, "Should wait for the full timeout")
+	assert.Greater(t, counter.Load(), int32(1), "Should have checked condition multiple times")
+}
+
+func TestEventually_VeryShortTimeout(t *testing.T) {
```

Comment:
This test looks a bit odd: what is "very short"? I guess it depends on where it runs. I was also wondering if there should an opposite case where func returns true, but I'm unsure.

Not wanting to say anything specific with this comment, just a brain dump :)

Reply:
I think very short here means that the timers are roughly on the same order of magnitude as the code execution time. Basically a sanity check for synchronization issues that may be hidden by larger timers. That being said, I was already on the fence for removing this test. So maybe let's just do that, because I agree that the test purpose is a bit vague.